src/utils.ts

```diff
@@ -525,3 +532,74 @@ export const defaultConvertTaskResultForConsoleTable: ConsoleTableConverter = (
     }),
   }
 }
+
+interface PLimitQueueItem {
+  fn: () => Promise<unknown>
+  reject: (error: unknown) => void
+  resolve: (value: unknown) => void
+}
+
+/**
+ * Creates a concurrency limiter that can execute functions with a maximum concurrency limit.
+ * @param limit - Maximum number of concurrent executions
+ * @returns A function that accepts a function to execute and returns a promise
+ */
+export const pLimit = (limit: number): PLimitInstance => {
+  const queue: PLimitQueueItem[] = []
+
+  let activeCount = 0
+  let pendingCount = 0
+
+  const processNext = (): void => {
+    while (activeCount < limit && queue.length > 0) {
+      const item = queue.shift()
```

Comment:
cant we use pop() instead?

IIRC shift() is basically creating a new Array under the hood and thus slower.